### PR TITLE
Adding release notes for 2.0 GA

### LINKED ISSUE #385: Release version 2.0.0 (GA)
This is a component issue for 2.0.0 (GA).
Coming from [opensearch-build 2086](https://github.com/opensearch-project/opensearch-build/issues/2086). Please follow the following checklist.
__Please refer to the DATES in that post__.

<details><summary>How to use this issue</summary>
<p>

## This Component Release Issue

This issue captures the state of the OpenSearch release, on component/plugin level; its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help. 
Any release related work can be linked to this issue or added as comments to create visiblity into the release status.

## Release Steps

There are several steps to the release process; these steps are completed as the whole component release and components that are behind present risk to the release. The component owner resolves the tasks in this issue and communicate with the overall release owner to make sure each component are moving along as expected.

Steps have completion dates for coordinating efforts between the components of a release; components can start as soon as they are ready far in advance of a future release. The most current set of dates is on the overall release issue linked at the top of this issue.

## The Overall Release Issue

Linked at the top of this issue, the overall release issue captures the state of the entire OpenSearch release including references to this issue, the release owner which is the assignee is responsible for communicating the release status broadly.  Please contact them or @mention them on that issue for help.

## What should I do if my plugin isn't making any changes?

If including changes in this release, increment the version on `2.0` branch to `2.0.0` for Min/Core, and `2.0.0.0` for components. Otherwise, keep the version number unchanged for both.

</p>
</details>

### Preparation

- [x] Assign this issue to a release owner.
- [x] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [x] All the tasks in this issue have been reviewed by the release owner.
- [x] Create, update, triage and label all features and issues targeted for this release with `v2.0.0`.

### CI/CD

- [x] All code changes for `2.0.0` are complete.
- [x] If this repo is core repo, use `2.0.0` version for your component; if this repo is plugin repo, use `2.0.0.0` version for your component. Qualifier should be empty `''` see this issue for details: https://github.com/opensearch-project/opensearch-build/issues/2078. 
- [x] Ensure working and passing CI.
- [x] Check that this repo is included in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.0.0).

### Pre-Release

- [x] Update to the `2.0` release branch in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.0.0).
- [x] Increment the version on the parent branch to the next development iteration.
- [x] Gather, review and publish release notes following the [rules](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md) and back port it to the release branch.[git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [x] Confirm that all changes for `2.0.0` have been merged.

### Release Testing

- [x] Find/fix bugs using latest tarball and docker image provided in parent release issue and update the release notes if necessary.
- [x] Code Complete: Test within the distribution, ensuring integration, backwards compatibility, and performance tests pass.
- [x] Sanity Testing: Sanity testing *and* fixing of critical issues found.
- [x] File issues for all intermittent test failures.

### Release

- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [x] Verify all issued labeled for this release are closed or labelled for the next release.

### Post Release

- [x] Prepare for an eventual security fix development iteration by incrementing the version on the release branch to the next eventual patch version.
- [x] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) of the next eventual security patch version.
- [x] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/blob/main/meta/templates/releases/release_template.md).
- [x] Conduct a retrospective, and publish its results.

### PR DESCRIPTION
**Pull request by martin-gaievski**
Signed-off-by: Martin Gaievski <gaievski@amazon.com>

### Description
Add release notes for 2.1.0 rc1
 
### Issues Resolved
#385 
 
### Check List
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Shivamdhar — DISMISSED** [review]

File: release-notes/opensearch-knn.release-notes-2.0.0.0.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Version 2.0.0.0 Release Notes
+
+Compatible with OpenSearch 2.0.0
+
+### Enhancements
+
+* Add Abstraction for BWC tests with a basic test ([#388](https://github.com/opensearch-project/k-NN/pull/388))
+* Add restart upgrade BWC Tests ([#387](https://github.com/opensearch-project/k-NN/pull/387))
+
+### Bug Fixes
```
Comment (naveentatikonda) [inline release-notes/opensearch-knn.release-notes-2.0.0.0.md:10]: Should we remove these headings if we don't have any PRs to add in that section?

File: release-notes/opensearch-knn.release-notes-2.0.0.0.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+### Infrastructure
+
+* Update jacoco version to 0.8.7 to support JDK 17  ([#372](https://github.com/opensearch-project/k-NN/pull/372))
+* Remove rc1 build qualifier for 2.0 GA release ([#395](https://github.com/opensearch-project/k-NN/pull/395))
+
+### Refactoring
```
Comment (naveentatikonda) [inline release-notes/opensearch-knn.release-notes-2.0.0.0.md:17]: Same here

File: release-notes/opensearch-knn.release-notes-2.0.0.0.md (hunk 1 above)
Comment (martin-gaievski) [reply release-notes/opensearch-knn.release-notes-2.0.0.0.md:10]: I had the same question, my reason to leave those in the notes - it's clean indication that we do not have any changes for that category. Does it make sense?

Comment (martin-gaievski) [reply release-notes/opensearch-knn.release-notes-2.0.0.0.md:10]: Ok, checked previous [notes](https://github.com/opensearch-project/k-NN/blob/main/release-notes/opensearch-knn.release-notes-1.1.0.0.md), seems that in general rule is - no change no category. Let me remove empty headings

Comment (naveentatikonda) [reply release-notes/opensearch-knn.release-notes-2.0.0.0.md]: I got your point. But, in all the previous release notes we are only adding the sections that contains PRs. Just want to make things look uniform.

Comment (martin-gaievski) [reply release-notes/opensearch-knn.release-notes-2.0.0.0.md]: Sure, I agree. Just pushed commit that corrects this, please check

**Review by naveentatikonda — APPROVED** [review]

**Review by Shivamdhar — APPROVED** [review]

**Pull request merged**